child_compassion/model/child_pictures.py

```diff
@@ -77,8 +77,43 @@ def create(self, cr, uid, vals, context=None):
             # We could not retrieve a picture, we cancel the creation
             self.unlink(cr, uid, res_id, context)
             return False
+        child_picture = self.browse(
+            cr, uid, res_id, context)
+        same_picture_ids = self._find_same_picture(
+            cr, uid, child.id,
+            child_picture.fullshot, child_picture.headshot,
+            context)
+        same_picture_ids.remove(child_picture.id)
+
+        if same_picture_ids:
+            self._unlink_related_attachment(cr, uid, child_picture.id, context)

```

Comment:
Nice method, you can call it as well after line 77, in order to remove attachments in case of failure.